### ISSUE TITLE
ImapClient onCommandResult fails with "type 'Null' is not a subtype of type 'FutureOr<Mailbox>'"

### ISSUE THREAD

**Issue by azulli** [Enough-Software/enough_mail#164]
Hi,
I've spent some time trying to figure out the source of this error, and I found out a quirk on the handling of the onCommandResult future completion related to specific mailbox commands.
When executing commands like CREATE or RENAME, any outcome fails with the "type X is not a subtype Y" error when the result attribute of ImapResponse is null.
This failure is due to the missing nullable modifier on the generic type of the CommandTask completer. 
When an IMAP mailbox command, like RENAME, completes successfully and a mailbox was previously selected, the error does not get triggered because the underlying SelectParser has been initialized with the current selected mailbox.

The following is a test case that shows the issue:
```dart
@Timeout(Duration(minutes: 45))

import 'package:enough_mail/imap/imap_client.dart';
import 'package:enough_mail/imap/mailbox.dart';
import 'package:test/test.dart';

// TODO: Set the IMAP connection parameters
final imapHost = '';
final imapPort = 993;
final imapUser = '';
final imapPass = '';

// Example fileds
final boxPrefix = 'INBOX.';
final boxName = 'oldname';
final newName = 'newname';
final boxFlags = <MailboxFlag>[];

void main() {
  test('NoopParser error in rename', () async {
    final client = ImapClient(isLogEnabled: true);
    var connected = false;
    Mailbox? box2;
    try {
      await client.connectToServer(imapHost, imapPort);
      connected = true;
      await client.login(imapUser, imapPass);
      // Enabling this avoids the error when successfully completed. Can be any mailbox
      // await client.selectInbox();
      var box = Mailbox.setup(boxName, '$boxPrefix$boxName', boxFlags);
      box2 = await client.renameMailbox(box, '$boxPrefix$newName');
      await client.logout();
    } catch (e, st) {
      print(e);
      print(st);
    } finally {
      if (connected) await client.disconnect();
    }
    expect(box2, isNotNull);
    expect(newName, box2?.name);
  });
}
```

**Comment by robert-virkus** [comment]
relates to #163 #160

**Comment by robert-virkus** [comment]
Thanks a lot for digging into this. I think I can fix that now! :-)

**Comment by azulli** [comment]
Glad I've helped.
As a side note, maybe the NoopParser could use the passed-in mailbox instead of _selectedMailbox for the rename command?

**Comment by robert-virkus** [comment]
Released fix with v1.3.4
